### DISCUSSION TITLE
Supported Python versions?

### DISCUSSION THREAD

**Discussion by leroyvn** [radis/radis#639] [General]
Hi all, I see on the [PyPI page](https://pypi.org/project/radis/) that v0.14 is tagged with Python 3.7 to 3.9 support. I have two questions:

* Python 3.9 is currently the oldest maintained version of Python, and Python 3.10 to 3.12 are maintained as well. Is there a particular reason to cap the supported Python version to 3.9?
* Python 3.7 is EOL, and Python 3.8 will be by the end of this year. These versions are no longer supported by Numpy maintainers, meaning that supporting these might implicitly prevent people depending on Radis to upgrade their Numpy-based dependencies (e.g. xarray or pandas). Is there a particular reason to still support these?

**Comment by mjrodri** [comment]
The decision to cap the supported Python version and continue supporting older versions of Python in a library or package is typically made by the maintainers based on various considerations. While I don't have specific information about the package you're referring to, I can provide some general reasons why maintainers might choose to support older Python versions:

User Base: If the package has a significant user base still relying on older Python versions, the maintainers may choose to support those versions to accommodate existing users and avoid breaking their workflows.

Compatibility: Supporting a broader range of Python versions ensures compatibility with different environments and dependencies that users may have. However, this can become challenging as newer Python versions introduce changes and improvements.

Dependency Compatibility: If the package relies on dependencies like Numpy, which may still support older Python versions, the maintainers might choose to align their package's support accordingly.

Transition Period: Maintainers may provide support for older Python versions during a transition period to give users time to update their environments and dependencies gradually.

However, it's also important for maintainers to consider the trade-offs. Supporting older Python versions can hinder the adoption of new features and improvements in the language. Additionally, it may increase the maintenance burden on the development team.

If you are concerned about the version support and its impact on dependencies like xarray or pandas, you may want to raise the question or discuss it with the package maintainers. Check the package's documentation, GitHub repository, or other communication channels for any discussions or announcements related to version support decisions. If needed, you can contribute to the discussion or open an issue expressing your concerns and suggesting potential changes.

**Comment by leroyvn** [comment]
Just to put some context around my question: I'm using RADIS as a dependency in another project, alongside many other dependencies. My issue is that all dependencies's Python and library version requirements propagate to my project, and this can lead to severe incompatibilities in some cases. In particular, RADIS supports Python 3.8 and 3.9, and many other dependencies support Python 3.9-3.12 in their most recent releases. This can lead to heavily constrained dependency systems that can be hard to resolve nicely.

So my underlying question is: are there plans to support Python 3.10+ in the near future?

**Reply by erwanp** [reply]
Hello, we are in the process of releasing a new version (with large updates), and it will support up to Python 3.12. 
Do you have a strong deadline on your side ?

**Reply by leroyvn** [reply]
Hi @erwanp, this is great news :) No hard deadline on my end.
